From: "Laszlo Boszormenyi (GCS)" <gcs@debian.org>
Date: Sun, 24 Nov 2024 23:29:27 -0800
Subject: limit block_size to 32 bit

Forwarded: no
Last-Update: 2021-10-16

Make the block_size to 3 Gb only on i386 and mipsel.
---
 db/db_test.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/db/db_test.cc b/db/db_test.cc
index ba65a7b..0d614ed 100644
--- a/db/db_test.cc
+++ b/db/db_test.cc
@@ -7350,7 +7350,11 @@ TEST_F(DBTest, LargeBlockSizeTest) {
   CreateAndReopenWithCF({"pikachu"}, options);
   ASSERT_OK(Put(0, "foo", "bar"));
   BlockBasedTableOptions table_options;
+#if !defined(__i386__) && !defined(__mips__)
   table_options.block_size = 8LL * 1024 * 1024 * 1024LL;
+#else
+  table_options.block_size = 3LL * 1024 * 1024 * 1024LL;
+#endif
   options.table_factory.reset(NewBlockBasedTableFactory(table_options));
   ASSERT_NOK(TryReopenWithColumnFamilies({"default", "pikachu"}, options));
 }
